pkg/cover/file.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package cover
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/civil"
+	"github.com/google/syzkaller/pkg/covermerger"
+)
+
+type lineRender func(string, bool, int, int) string
+
+func RendFileCoverage(c context.Context, ns, repo, commit, filePath string,
+	fromDate, toDate civil.Date, render lineRender) (string, error) {
+	fileContent, errGetFileVersion := covermerger.GetFileVersion(filePath, repo, commit)
+	if errGetFileVersion != nil {
+		return "", fmt.Errorf("failed to GetFileVersion for file %s, commit %s from repo %s: %w",
+			filePath, commit, repo, errGetFileVersion)
+	}
+	config := &covermerger.Config{
+		Jobs: 1,
+		Base: covermerger.RepoBranchCommit{
+			Repo:   repo,
+			Branch: "",
+			Commit: commit,
+		},
+		FileVersProvider: covermerger.MakeWebGit(),
+	}
+
+	dbReader := covermerger.MakeBQCSVReader()
+	if err := dbReader.InitNsRecords(c,
+		ns,
+		filePath,
+		fromDate,
+		toDate,
+	); err != nil {
+		return "", fmt.Errorf("failed to dbReader.InitNsRecords: %w", err)
+	}
+	defer dbReader.Close()
+	csvReader, errReader := dbReader.Reader()
+	if errReader != nil {
+		return "", fmt.Errorf("failed to dbReader.Reader: %w", errReader)
+	}
+
+	mergeResult, errMerge := covermerger.MergeCSVData(config, csvReader)
```

Comment:
s/errMerge/err/

Reply:
I started to hate the "err" usage. Calling the next function it is often a question, was err defined or not, is it the only return parameter of the next function or not.

Reply:
What are you trying to do that does not work?

Usually it's not a problem:
https://go.dev/play/p/9BpBGp7T1EL

Reply:
The last "if err := qux(); err != nil {".
It works and looks good.
The main problem - my IDE complains about shadowing and it's distracting me.

Reply:
Disabling shadowing detection for "err" should solve the problem.

Reply:
My IDE can't opt out variables  by name :(.
I'll live with it... (
Fixed.